doc/nrf/releases/release-notes-2.1.0.rst

```diff
@@ -0,0 +1,1030 @@
+=======
+
+The MCUboot fork in |NCS| (``sdk-mcuboot``) contains all commits from the upstream MCUboot repository up to and including ``1d4404116a9a6b54d54ea9aa3dd2575286e666cd``, plus some |NCS| specific additions.
+
+The code for integrating MCUboot into |NCS| is located in the :file:`ncs/nrf/modules/mcuboot` folder.
+
+The following list summarizes both the main changes inherited from upstream MCUboot and the main changes specific to the |NCS|:
+
+* boot_serial:
+
+  * Upgraded from cddl-gen v0.1.0 to zcbor v0.4.0.
+  * Refactored and optimized the code, mainly in what affects the progressive erase implementation.
+  * Fixed a compilation issue with the echo command code.
+
+* imgtool: Added support for providing signature through a third party.
+* zephyr:
+
+  * Added initial support for leveraging the RAM-LOAD mode with the zephyr-rtos port.
+  * Added the MCUboot status callback support.
+    See :kconfig:option:`CONFIG_MCUBOOT_ACTION_HOOKS`.
+  * Edited includes to have the ``zephyr/`` prefix.
+  * Edited the DFU detection's GPIO-pin configuration to be done through DTS using the ``mcuboot-button0`` pin alias.
+  * Edited the LED usage to prefer DTS' ``mcuboot-led0`` alias over the ``bootloader-led0`` alias.
+  * Removed :c:func:`device_get_binding()` usage in favor of :c:func:`DEVICE_DT_GET()`.
+
+* Documentation:
+
+  * :ref:`mcuboot:mcuboot_ncs` now includes information regarding the bootloader user guides in the |NCS| documentation.
+  * Removed:
+
+    * The "Zephyr Test Plan" page in the MCUboot documentation set.
+    * The "Building and using MCUboot with Zephyr" page in the MCUboot documentation set.
+
+Zephyr
+======
+
+.. NOTE TO MAINTAINERS: All the Zephyr commits in the below git commands must be handled specially after each upmerge and each NCS release.
+
+The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``71ef669ea4a73495b255f27024bcd5d542bf038c``, plus some |NCS| specific additions.
+
+For the list of upstream Zephyr commits (not including cherry-picked commits) incorporated into nRF Connect SDK since the most recent release, run the following command from the :file:`ncs/zephyr` repository (after running ``west update``):
+
+.. code-block:: none
+
+   git log --oneline 71ef669ea4 ^45ef0d2
+
+For the list of |NCS| specific commits, including commits cherry-picked from upstream, run:
+
+.. code-block:: none
+
+   git log --oneline manifest-rev ^71ef669ea4
+
+The current |NCS| main branch is based on revision ``71ef669ea4`` of Zephyr.
+
+The following list summarizes the major additions specific to the |NCS|:
+
+* Added Wi-Fi L2 layer.
+  This is a fork of Zephyr's Wi-Fi L2 with the added support for the WPA supplicant.
+
+zcbor
+=====
+
+* Updated the `zcbor`_ module from v0.4.0 to v0.5.1.
+  Release notes for v0.5.0 and v0.5.1 are located at :file:`ncs/modules/lib/zcbor/`.
+* Regenerated :ref:`lib_fmfu_fdev` code using zcbor v0.5.1.
+
+Trusted Firmware-M
```

Comment:
Added:
    TF-M support for FPU Hard ABI.

Updated:
    Updated to TF-M 1.6.0.

Fixed:
    An issue with Thingy:91 v1.5.0 and lower version.
    An issue where TF-M used more RAM compared to SPM in the minimal configuration
    An issue with non-secure storage partitions in external flash.